commands/ls.go

```diff
@@ -232,11 +233,16 @@ func matchesName(host *host.Host, names []string) bool {
 func attemptGetHostState(h *host.Host, stateQueryChan chan<- HostListItem) {
 	url := ""
 	hostError := ""
+	dockerVersion := "?"
 
 	currentState, err := h.Driver.GetState()
 	if err == nil {
 		url, err = h.GetURL()
 	}
+	if err == nil {
+		dockerVersion, err = h.GetDockerVersion()
```

Comment:
If the Docker daemon is not running, won't this bit hang?  Why not wrap this code in its own timeout to avoid the catch-all timeout falsely reporting state as `state.Timeout` if it's actually `state.Running`?

Likewise, what will happen here if `state` is `state.Stopped`?  Will it hang?